debian/postinst

```diff
@@ -68,6 +68,10 @@ hildon-application-manager-config update || true
 ## Clean HAM screenshot
 rm -f /home/user/.cache/launch/com.nokia.hildon_application_manager.pvr
 
+## HACK
+mkdir -p /usr/libexec
+ln -s /usr/lib/$(dpkg-architecture -q DEB_HOST_MULTIARCH)/apt-worker /usr/libexec/apt-worker
```

Comment:
I think it should be done using configure.ac with AC_SUBST and friends

Reply:
Could you help me with this please? I'm not well versed with autoconf.

Reply:
I think this should be done by using https://wiki.debian.org/Multiarch/Implementation#Dynamic_debian.2F.2A_files

Reply:
@freemangordon The thing is that the path is hardcoded throughout h-a-m.

```
$ grep -RnisI libexec
ChangeLog:5678: * src/Makefile.am: Removed dist_libexec_SCRIPTS, since the
ChangeLog:5681: * src/main.cc: Use /usr/libexec/apt-worker.bin as the apt-worker
Makefile.am:18:rescuedir = $(libexecdir)
debian/hildon-application-manager.launcher:2:/usr/libexec/ham-after-boot
debian/hildon-application-manager.xsession:2:/usr/libexec/ham-after-boot &
debian/postinst:40:rem_inittab "resc::bootwait:/usr/libexec/apt-worker.bin rescue"
debian/postinst:44:rem_inittab "resc::bootwait:/usr/libexec/apt-worker rescue"
debian/postinst:72:mkdir -p /usr/libexec
debian/postinst:73:ln -s /usr/lib/$(dpkg-architecture -q DEB_HOST_MULTIARCH)/apt-worker /usr/libexec/apt-worker
debian/postrm:3:rm -f /usr/libexec/apt-worker
ham-rescue.sh:53:/usr/libexec/apt-worker rescue
hildon-application-manager.sudoers:1:%users ALL = NOPASSWD: /usr/libexec/apt-worker
src/Makefile.am:7:libexec_PROGRAMS = apt-worker ham-after-boot
src/apt-worker-client.cc:46:#define APT_WORKER_CMD_DEFAULT "/usr/libexec/apt-worker"
statusbar/ham-updates.c:457:    "/usr/libexec/apt-worker",
```

Reply:
Guess I'll just learn how to substitute it with autoconf :)

Reply:
If you will have troubles with autoconf (I think you will), the mandatory thing here is to move this part out from debian/ directory. It should be done somewhere in Makefiles. So the package won't depend on Debian build system to be installed correctly.

Reply:
Perhaps `ham-after-boot` and `apt-worker` should just be installed in `$(PREFIX)/bin`? If you take a look at any current debian-based system, there is never anything in `/usr/libexec`

Reply:
Mitigated with e42f64940d62f34f87f7e0b334376bfb59a0f447

Reply:
> Perhaps `ham-after-boot` and `apt-worker` should just be installed in `$(PREFIX)/bin`? If you take a look at any current debian-based system, there is never anything in `/usr/libexec`

I have a better option for you: take a look at how it's done in upower.
It hides its binary that shouldn't be executed by the user in /usr/lib/upower/.
It's a good example of how this should be done. Maybe we can do the same for h-a-m?

Reply:
I understand, but I am not sure how to replace the hardcoded strings in the c files.

Reply:
Just replace the hardcoded values in the sources by hand.
And do a separate commit, please :)